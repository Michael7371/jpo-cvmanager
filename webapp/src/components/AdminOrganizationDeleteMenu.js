import React from 'react'
import { confirmAlert } from 'react-confirm-alert'
import DeleteIcon from '@mui/icons-material/Delete'
import 'react-confirm-alert/src/react-confirm-alert.css'
import { Options } from './AdminDeletionOptions'

import '../features/adminRsuTab/Admin.css'

const AdminOrganizationDeleteMenu = (props) => {
  const handleDelete = () => {
    const buttons = [
      {
        label: 'Yes',
        onClick: () => props.deleteOrganization(),
      },
      { label: 'No', onClick: () => {} },
    ]
    const alertOptions = Options(
      'Delete Organization',
      'Are you sure you want to delete the "' + props.selectedOrganization + '" organization?',
      buttons
    )
    confirmAlert(alertOptions)
  }

  return (
    <div>
      <button className="delete_button" onClick={handleDelete} title="Delete Organization">
        <DeleteIcon size={20} />
      </button>
    </div>
  )
}

export default AdminOrganizationDeleteMenu
