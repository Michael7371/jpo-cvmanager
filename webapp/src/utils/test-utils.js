/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
