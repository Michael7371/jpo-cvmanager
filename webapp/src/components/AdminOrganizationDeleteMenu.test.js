import React from 'react'
import { render } from '@testing-library/react'
import AdminOrganizationDeleteMenu from './AdminOrganizationDeleteMenu'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<AdminOrganizationDeleteMenu />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
