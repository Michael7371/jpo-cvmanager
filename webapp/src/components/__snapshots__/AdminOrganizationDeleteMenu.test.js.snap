// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div>
    <button
      class="delete_button"
      title="Delete Organization"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        size="20"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </button>
  </div>
</div>
`;
